Applications/FileIO/CsvInterface.h

```diff
@@ -36,9 +39,36 @@ namespace FileIO {
 /**
  * Interface for reading CSV file formats.
  */
-class CsvInterface {
+class CsvInterface  : public BaseLib::IO::Writer
+{
 
 public:
+    /// Contructor (only needed for writing files)
+    CsvInterface() {};
+
+    /// Adds an index vector of size s to the CSV file
+    void addIndexVectorForWriting(std::size_t s);
+
+    /// Adds a data vector to the CSV file. All data vectors have to have the same size.
+    /// Vectors will be written in the same sequence they have been added to the interfaceW.
+    template<typename T>
+    bool addVectorForWriting(std::string const& vec_name, std::vector<T> const& vec)
+    {
+        if (typeid(vec) == typeid(std::vector<std::string>) ||
+            typeid(vec) == typeid(std::vector<double>) ||
+            typeid(vec) == typeid(std::vector<int>))
```

Comment:
You check that at runtime is there a reason why you don't do it at compile time?